### ISSUE TITLE
GNOME 43.3 - Windows are not automatically send to the appropriate workspaces

### ISSUE THREAD

**Issue by zDEFz** [khimaros/smart-auto-move#38]
How to reproduce:
1. Have at least 2 static workspaces
2. Add keepassxc in the first or the second workspace
3. Add thunderbird in the first or the second workspace
4. Make both fullscreen

Signing in then, the user has to click into the GNOME desktop and then choose for the window, either keepassxc or thunderbird, which is then immediately full-screened. But its now two fullscreenwindows in one workspace.

How to solve:
Toggle Maximization state of the window thats meant to be in another workspace.

Things I tried todo: 
- Delete all states in the extension and just save with non-fullscreen windows.

Result: 
- Windows are still opening up to take the entire space.

Video: Coming

**Comment by zDEFz** [comment]
After fiddling with it a lot, I learned on how to use the software. Restoring is really dependent on application, on window state and on match Treshhold. I had to relog a lot in combination with the autostarts extension to see if it really worked. But now that it does, I'm happy.

**Comment by zDEFz** [comment]
Fix is essentially that some applications wont be moved no matter what and others are. So you can get some via flatpak and the others you can run natively. And then when something still doesnt work, you gotta remove the saved windows and readd them overriding them. I think I relogged about 400 times for my 8 workspaces with many apps and did tons of other tweaking. But now when I log in, its restored as I left it :)

**Comment by khimaros** [comment]
@zDEFz thank you for the detailed issue description and trying to make this work. was it still operating unreliably for you even once you figured out the right set of overrides?

**Comment by zDEFz** [comment]
> @zDEFz thank you for the detailed issue description and trying to make this work. was it still operating unreliably for you even once you figured out the right set of overrides?

Yes. I tried everything. This was the reason I'm back to xorg / i3 and wayland / sway

**Comment by khimaros** [comment]
i've released https://github.com/khimaros/smart-auto-move/releases/tag/v19 which contains commit https://github.com/khimaros/smart-auto-move/commit/dea6f57389a4dba30f5200af91134ce43ada28d9 -- there are some changes to the fullscreen and workspace management which may help with this issue.
